### PR TITLE
Hack script can install published cartographer release

### PR DESCRIPTION
**Pull request by waciumawanjohi**
[PR requirements]: https://github.com/vmware-tanzu/cartographer/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines

## Changes proposed by this PR

The hack script can be useful for teaching users about cartographer. But this script installs the version of Cartographer from whatever branch a user happens to be on (for learners, presumably this will be main). This commit provides an option for users to install the latest version of Cartographer to their spun up cluster:

```
./hack/setup.sh cluster cartographer-latest
```

## Release Note

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- ~[ ] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`~
- [x] Removed non-atomic or `wip` commits
- ~[ ] Filled in the [Release Note](#Release-Note) section above~
- ~[ ] Modified the docs to match changes~

### CODE REVIEW

File: hack/setup.sh [hunk 1]
```diff
@@ -37,6 +37,7 @@ readonly KPACK_VERSION=0.5.1
 readonly SECRETGEN_CONTROLLER_VERSION=0.6.0
 readonly SOURCE_CONTROLLER_VERSION=0.17.0
 readonly TEKTON_VERSION=0.30.0
+readonly CARTOGRAPHER_LATEST_VERSION=$(curl -sL https://api.github.com/repos/vmware-tanzu/cartographer/releases/latest | jq -r ".tag_name")
```
Comment (cirocosta) [inline hack/setup.sh]: I think it's not a biiig big deal, but I'd personally prefer to not have http requests in the definition of a variable

e.g., a `--help` would involve that request being made:

```
bash -x ./hack/setup.sh --help 2>&1 | grep curl
++ curl -sL https://api.github.com/repos/vmware-tanzu/cartographer/releases/latest
```

given that is not really a constant like the other ones, do you think we could move this down to the `install_cartographer_latest_release` function? that way we only make the request if the person is intending to install in such way

Comment (cirocosta) [reply hack/setup.sh]: or actually, given that we can't even replace `CARTOGRAPHER_LATEST_VERSION` with an environment variable, I think we could instead just go with `https://github.com/vmware-tanzu/cartographer/releases/latest/download/cartographer.yaml` ?

e.g.

```console
$ kapp tools inspect -f https://github.com/vmware-tanzu/cartographer/releases/latest/download/cartographer.yaml

Resources in HTTP URL 'https://github.com/vmware-tanzu/cartographer/releases/latest/download/cartographer.yaml'

Namespace            Name                                  Kind                            Owner  Conds.  Rs  Ri  Age
(cluster)            cartographer-cluster-admin            ClusterRoleBinding              -      -       -   -   -
^                    cartographer-controller-admin         ClusterRole                     -      -       -   -   -
^                    cartographer-system                   Namespace                       -      -       -   -   -
^                    cartographer-user-admin               ClusterRole                     -      -       -   -   -
^                    cartographer-user-view                ClusterRole                     -      -       -   -   -
^                    clusterconfigtemplates.carto.run      CustomResourceDefinition        -      -       -   -   -
^                    clusterdeliveries.carto.run           CustomResourceDefinition        -      -       -   -   -
^                    clusterdeploymenttemplates.carto.run  CustomResourceDefinition        -      -       -   -   -
^                    clusterimagetemplates.carto.run       CustomResourceDefinition        -      -       -   -   -
^                    clusterruntemplates.carto.run         CustomResourceDefinition        -      -       -   -   -
^                    clustersourcetemplates.carto.run      CustomResourceDefinition        -      -       -   -   -
^                    clustersupplychains.carto.run         CustomResourceDefinition        -      -       -   -   -
^                    clustersupplychainvalidator           ValidatingWebhookConfiguration  -      -       -   -   -
^                    clustertemplates.carto.run            CustomResourceDefinition        -      -       -   -   -
^                    deliverables.carto.run                CustomResourceDefinition        -      -       -   -   -
^                    deliveryvalidator                     ValidatingWebhookConfiguration  -      -       -   -   -
^                    runnables.carto.run                   CustomResourceDefinition        -      -       -   -   -
^                    workloads.carto.run                   CustomResourceDefinition        -      -       -   -   -
cartographer-system  cartographer-controller               Deployment                      -      -       -   -   -
^                    cartographer-controller               ServiceAccount                  -      -       -   -   -
^                    cartographer-webhook                  Certificate                     -      -       -   -   -
^                    cartographer-webhook                  Secret                          -      -       -   -   -
^                    cartographer-webhook                  Service                         -      -       -   -   -
^                    private-registry-credentials          Secret                          -      -       -   -   -
^                    selfsigned-issuer                     Issuer                          -      -       -   -   -

Rs: Reconcile state
Ri: Reconcile information

25 resources

Succeeded
```

Comment (waciumawanjohi) [reply hack/setup.sh]: I like it. Updated to adopt the suggestion.

**Review by martyspiewak — APPROVED** [review]
Lgtm!

**Pull request merged**